### ISSUE TITLE
Improving accessibility on Alert and Modals

### ISSUE THREAD

**Issue by andresgalante** [twbs/bootstrap#23664]
I was wondering if we should change `role="alert"` to `role="alertdialog"` on [dismissible alerts](https://getbootstrap.com/docs/4.0/components/alerts/#dismissing) since W3C alert role specs read:
>If an alert requires focus to close the alert, then content authors SHOULD use alertdialog instead.
https://www.w3.org/TR/wai-aria/roles#alert

And according to MDN on `role=“alert”`:
> If an alert also provides interactive controls (such as form controls that allow the user to rectify a problem, or an "OK" button that discards the alert) the `alertdialog` role should be used instead. 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role

This would mean wrapping the text with a div with an id and adding an `aria-describedby` which I don't quite like but on modals we have plenty of hooks to make it more accessible.

This was also mention here #22429

What do you think? Should I introduce this idea as a PR?

**Comment by patrickhlauke** [comment]
agree that for dismissible alerts, `alertdialog` role is more appropriate. would also be good to add a sentence or two in the documentation pointing this out.

**Comment by andresgalante** [comment]
@patrickhlauke Cool, I'll send a PR today and tag you

**Comment by andresgalante** [comment]
@pablocrivella Let's rethink this. After re reading:
https://www.w3.org/TR/wai-aria/roles#alertdialog

I don't think that:
>Content authors SHOULD make alert dialogs modal by ensuring that, while the alertdialog is shown, keyboard and mouse interactions only operate within the dialog.

Is not accurate to an alert since the rest of the page should still be clickable, an alert is definitely not a modal.

I've also read strong opinion against using `alertdialog`:
https://www.marcozehe.de/2015/02/05/advanced-aria-tip-2-accessible-modal-dialogs/

Should we just close this issue for now?

**Comment by bkdotcom** [comment]
@andresgalante 
a modal with 
`data-backdrop="static"`
could be considered an alert as clicking outside the "static" modal does not dismiss the modal

**Comment by pablocrivella** [comment]
@andresgalante did you mean to mention @patrickhlauke?

**Comment by andresgalante** [comment]
Yeap. sorry :( autofill kills me sometimes, sorry Pablo.
